crates/nodeapi/src/lib.rs

```diff
@@ -64,14 +63,83 @@ impl NodeApi {
         }
     }
 
+    /// Add a service to the node
+    pub fn register_service(
+        &mut self,
+        id: &str,
+        name: &str,
+        description: &str,
+        status: ServiceStatus,
+    ) {
+        self.services.push(NodeService {
+            id: id.to_string(),
+            name: name.to_string(),
+            description: description.to_string(),
+            status,
+        });
+    }
+
+    pub fn update_service_status(
+        &mut self,
+        service_id: &str,
+        status: ServiceStatus,
+    ) -> Result<(), NodeApiError> {
+        for service in self.services.iter_mut() {
+            if service.id == service_id {
+                service.status = status;
+                return Ok(());
+            }
+        }
+
+        Err(NodeApiError::ServiceIdNotFound(service_id.to_string()))
+    }
+
+    pub fn deregister_service(&mut self, service_id: &str) -> Result<(), NodeApiError> {
+        if let Some(index) = self.services.iter().position(|s| s.id == service_id) {
+            self.services.remove(index);
+            return Ok(());
+        }
+        Err(NodeApiError::ServiceIdNotFound(service_id.to_string()))
+    }
+}
+
+#[derive(Clone)]
+pub struct NodeApi(Arc<Mutex<NodeInfo>>);
+
+impl NodeApi {
+    /// Function to create the Ntex HTTP server
+    /// This function sets up the server and routes.
+    /// External users can call this function to create and run the server.
+    pub fn create_server(
+        api: Arc<Mutex<NodeInfo>>,
+        ip_port_addr: String,
+    ) -> std::io::Result<ntex::server::Server> {
```

Comment:
We should split this into two steps:

```rust
impl NodeApi {
    pub fn new(info: NodeInfo) -> Self { ... }
    pub fn start(ip_port_addr: String) -> std::io::Result<ntex::server::Server> { ... }
```

This way users don't need to manually wrap the `NodeInfo` into `Arc<Mutex<NodeInfo>>`. We also disallow calling methods on `Arc<Mutex<NodeInfo>>` directly (since we want them to go through `NodeApi`. We also keep the possibility of changing this to use `RwLock`, a channel, or even double buffers in the future.